src/components/locale-link.tsx

```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import { Link, GatsbyLinkProps } from "gatsby";
+import { useCurrentLocale } from "../util/use-locale";
+
+/**
+ * Like Gatsby's <Link>, but it prefixes the passed-in `to` prop with
+ * the current locale.
+ *
+ * Note that this doesn't localize the actual *text* of the link--it only localizes
+ * the path!
+ */
+export function LocaleLink<TState>(
+  props: GatsbyLinkProps<TState> & { ref?: never }
```

Comment:
I was having weird type issues with the `ref` prop but since we never actually use it in our code, I just simplified things by prohibiting it.